### PR TITLE
fhenixsdk (PermitV2)

### PR DESCRIPTION
**Pull request by architect-dev**
**Introduces the `PermitV2` class**, responsible for creating and signing a permit, extracting the permission, and using it to unseal data. It is serializable with the `.serialize` and `deserialize` methods.

**Introduces the `FhenixClientV2` class**, responsible for initializing FHEVM, managing fheKeys, connecting the user's wallet with the `account` / `send` / `signTypedData` functions, and interacting with localstorage through the new zustand store.

**Introduces a persisted (localstorage) zustand store `fhenixStore`** which contains `permits` a mapping from account -> permitHash -> permit, `activePermitHash` a mapping from account -> permitHash, and `fheKeys` a mapping from chainId -> securityZone -> FheKey. It uses the zustand persist middleware to interact with localstorage, which is well tested and reduces a lot of overhead on our side.

**Implementation:**
Users can interact with the PermitV2s directly by using the new PermitV2 class to manage permissions. The FhenixClientV2 is a utility layer for easing the use of permitV2s through things like `activePermitHash` which stores which permit the user is currently using. Since the PermitV2 system can handle multiple contracts, I expect most use cases will not require permit juggling (if it does this is easy to implement with a system built directly on top of the PermitV2 class).

I'm not thrilled with the contstruct/initialize api of the FhenixClientV2, this is what it looks like to set up the client:
```typescript
const client = new FhenixClientV2();
await client.initialize({
      account: bobAddress,
      send: bobProvider.send,
      signTypedData: bobSigner.signTypedData,
});
```

But if `account`, `send`, `signTypedData` change the class is simply re-initialized to update it, which is a pattern I like. Re-initializing will also pull existing data for the new account from the store/localstorage if it exists.

This is purely an addition to the existing `fhenix.js`, it doesn't introduce any api changes or breaking changes, and is effectively opt-in.

### CODE REVIEW

File: src/extensions/store/sdk.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    );
+  }
+
+  return chainId;
+};
+
+export type InitParams = {
+  account: string;
+  send: SendFn;
+  signTypedData: SignTypedDataFn;
```
Comment (Cashmaney) [inline src/extensions/store/sdk.ts:102]: Is this an intuitive interface? Knowing how to pass these methods as parameters sounds a bit challenging I would think

Comment (architect-dev) [reply src/extensions/store/sdk.ts:102]: The intent was to allow users to be provider agnostic, and to allow viem based "providers" (referred to as clients in viem) to interact with the sdk.

I'm happy to revert this, I'm not convinced that this is better than the existing version.

File: src/extensions/store/permits.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import { createStore } from "zustand/vanilla";
+import { persist } from "zustand/middleware";
```
[the remark this answers is not in the document]
Comment (architect-dev) [reply src/extensions/store/permits.ts:2]: Absolutely, it leverages the local storage of mobile browsers, and has reasonable fallbacks like sessionstorage

File: README.md [hunk 3]
```diff
@@ -115,6 +115,47 @@ Also, if you had to fiddle with a bundler or config to get it working, please sh
...
+
+// Create PermitV2 (will trigger a wallet signature in web3 frontend)
+await client.createPermit({
+  type: 'self',
+  issuer: account,
+  projects: ["COUNTER", "FHERC20"],
+})
+
+// Encrypt (will encrypt any nested encryptables)
+const encrypted = client.encryptTyped([
```
[the remark this answers is not in the document]
Comment (architect-dev) [reply README.md:137]: Agreed, I've since changed it in the actual sdk to simply be `encrypt`, but didn't update the README

**Comment by architect-dev** [comment]
In response to your comments @Cashmaney I've switched from the `send` + `signTypedData` interface back to an abstract provider / signer interface. I think in the end you're right that it's more intuitive for devs to pass providers / signers than pointers to individual functions. 

I've done a large pass to improve error messages and docstrings, I don't think they're perfect yet but I think they are good enough for now.

**Pull request merged**